### PR TITLE
[0.4.1] Fix: cli runner stream capture via capfd

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

Capturing of `sys.stdout` and `sys.stderr` via autouse fixture setting `capfd` on the `CliRunner` in `BaseCliTest`. Optional use of this patch was recently added in open-aea release v1.25.0 to reduce flakiness in tests.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: tests/test_autonomy/test_cli/test_deploy/test_from_token.py [hunk 1]
```diff
@@ -104,7 +103,6 @@ def setup(self) -> None:
 
     def test_from_token(
         self,
-        capsys: Any,
     ) -> None:
         """Run test."""
```
Comment (Karrenbelt) [inline tests/test_autonomy/test_cli/test_deploy/test_from_token.py:107]: `capfd` cannot be used in conjunction with `capsys`

File: tests/test_autonomy/test_cli/test_deploy/test_from_token.py [hunk 2]
```diff
@@ -137,12 +135,10 @@ def test_from_token(
...
-            out, err = capsys.readouterr()
-
-            assert result.exit_code == 0, out
-            assert "Service name: valory/oracle_hardhat" in out, err
-            assert "Building required images" in out, err
-            assert "Service build successful" in out, err
+            assert result.exit_code == 0, result.stdout
+            assert "Service name: valory/oracle_hardhat" in result.stdout
+            assert "Building required images" in result.stdout
+            assert "Service build successful" in result.stdout
```
Comment (Karrenbelt) [inline tests/test_autonomy/test_cli/test_deploy/test_from_token.py:141]: initialized with default parameters
https://github.com/valory-xyz/open-autonomy/blob/dc7748f4c2dd7a14f9b278c6aaeb59f734f3cba4/tests/test_autonomy/test_cli/base.py#L55
which means means that stderr is mixed with stdout ([reference](https://github.com/valory-xyz/open-autonomy/blob/dc7748f4c2dd7a14f9b278c6aaeb59f734f3cba4/tests/test_autonomy/test_cli/base.py#L55))

**Review by dvilelaf — APPROVED** [review]
LGTM

**Comment by Karrenbelt** [comment]
- failures e2e - the infamous timestamp is None issue
    ```
    2022-12-02T18:37:37.8534156Z [92492]stdout>  File "/tmp/tmp5hfwmdcx/agent_00000/vendor/valory/skills/abstract_round_abci/base.py", line 1992, in last_timestamp
    2022-12-02T18:37:37.8534324Z [92492]stdout>    raise ABCIAppInternalError("last timestamp is None")
    ```

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**